centreon/src/Core/AgentConfiguration/Application/UseCase/AddAgentConfiguration/AddAgentConfiguration.php

```diff
@@ -76,6 +77,20 @@ public function __invoke(
             }
 
             $request->pollerIds = array_unique($request->pollerIds);
+
+            if ($this->isCloudPlatform && ! $this->user->isAdmin()) {
+                $pollers = $this->readAcRepository->findPollersByIds($request->pollerIds);
+                foreach ($pollers as $poller) {
+                    if ($poller !== null && $poller->isLocalhost() === '1' && ! $poller->isRemoteServer()) {
```

Comment:
```suggestion
                    if ($poller->isLocalhost() === '1' && ! $poller->isRemoteServer()) {
```

According to your interface and your implementation $poller could not be null